### PR TITLE
Test support of DRF HyperlinkedIdentityField

### LINKED ISSUE #298: Support for HyperlinkedIdentityField
```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/snippets/

Django Version: 1.10.3
Python Version: 2.7.12
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'rest_framework_jwt',
 'rest_framework_json_api',
 'snippets.apps.SnippetsConfig']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback:

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  39.             response = get_response(request)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  217.                 response = self.process_exception_by_middleware(e, request)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  215.                 response = response.render()

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/django/template/response.py" in render
  109.             self.content = self.rendered_content

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework/response.py" in rendered_content
  72.         ret = renderer.render(self.data, accepted_media_type, context)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework/renderers.py" in render
  701.         context = self.get_context(data, accepted_media_type, renderer_context)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework/renderers.py" in get_context
  659.             'content': self.get_content(renderer, data, accepted_media_type, renderer_context),

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework/renderers.py" in get_content
  409.         content = renderer.render(data, accepted_media_type, renderer_context)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework_json_api/renderers.py" in render
  469.                     json_resource_obj = self.build_json_resource_obj(fields, resource, resource_instance, resource_name)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework_json_api/renderers.py" in build_json_resource_obj
  390.         relationships = JSONRenderer.extract_relationships(fields, resource, resource_instance)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework_json_api/renderers.py" in extract_relationships
  91.             relation_type = utils.get_related_resource_type(field)

File "/home/user/.pyenv/versions/djangorest/lib/python2.7/site-packages/rest_framework_json_api/utils.py" in get_related_resource_type
  202.                     parent_model_relation = getattr(parent_model, relation.field_name)

Exception Type: AttributeError at /snippets/
Exception Value: type object 'Snippet' has no attribute 'highlight'
```
Trying to integrate djangorestframework-jsonapi into the [official djangorestframework tutorial](https://github.com/tomchristie/rest-framework-tutorial), but I keep getting errors. I followed the [docs configuration instructions](https://django-rest-framework-json-api.readthedocs.io/en/stable/usage.html#configuration), and changed my `serializers.py` import to `from rest_framework_json_api import serializers`.  What am I missing here? Does djangorestframework-jsonapi not support the `HyperlinkedIdentityField`? Or should I be using the `ResourceRelatedField` for relationships? Any help is greatly appreciated!

**Side note:** _If there's anything I would improve upon it's the documentation. A tutorial, possibly forked from the official drf one, would be nice. Thank you for creating this project!_ :)

### PR DESCRIPTION
**Pull request by Alig1493**
Fixes #298

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [ ] documentation updated
- [ ] changelog entry added to `CHANGELOG.md`
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by sliverc** [comment]
It seems that HyperlinkedIdentityField doesn't support [source](https://github.com/encode/django-rest-framework/blob/master/rest_framework/relations.py#L434) attribute to be overwritten.

That's why an exception is raised. This should not concern us though as it is not correctly used. So best adjust your test case that we focus on simply using `HyperlinkedIdentityField` and see that it renders correctly.

**Comment by Alig1493** [comment]
@sliverc I rewrote the tests using default drf serializers to see if the entry works with hyperlinkedrelated field. No issue so far.

File: example/serializers.py [hunk 1]
```diff
@@ -173,6 +173,22 @@ class JSONAPIMeta:
...
 
 
+class EntryDRFSerializers(drf_serilazers.ModelSerializer):
+
+    tags = TaggedItemDRFSerializer(many=True, read_only=True)
+    url = drf_serilazers.HyperlinkedIdentityField(
+        view_name='drf-entry-blog-detail',
+        lookup_url_kwarg='entry_pk',
+        read_only=True,
+        source='blog'
```
Comment (sliverc) [inline example/serializers.py]: source attribute doesn't do anything so should be removed.

File: example/serializers.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ class Meta:
 
 class BlogSerializer(serializers.ModelSerializer):
     copyright = serializers.SerializerMethodField()
-    tags = TaggedItemSerializer(many=True, read_only=True)
+    tags = TaggedItemDRFSerializer(many=True, read_only=True)
```
Comment (sliverc) [inline example/serializers.py]: why has this changed? I guess we should stick to the DJA version here as previous.

File: example/views.py [hunk 3]
```diff
@@ -104,6 +105,22 @@ def get_object(self):
         return super(EntryViewSet, self).get_object()
 
 
+class DRFEntryViewSet(viewsets.ModelViewSet):
+    queryset = Entry.objects.all()
+    lookup_url_kwarg = 'entry_pk'
+
+    def get_serializer_class(self):
```
Comment (sliverc) [inline example/views.py]: easier defined as attribute:
`serializer_class = EntryDRFSerializers`

File: example/serializers.py (hunk 2 above)
Comment (Alig1493) [reply example/serializers.py]: Oh I thought it was under BlogDRFSerializer. Was careless.

**Comment by Alig1493** [comment]
@sliverc Made requested changes

**Comment by sliverc** [comment]
@Alig1493 Thanks.

**Pull request merged**